### PR TITLE
Use fixed user environment for tests

### PR DESCRIPTION
**Pull request by reinhapa**
### Description:
- converts the argument line into system property variables
- defines a fixed user environment for running the tests

### CODE REVIEW

**Review by dizzzz — APPROVED** [review]

**Review by dizzzz — APPROVED** [review]

**Comment by dizzzz** [comment]
image

**Review by dizzzz — APPROVED** [review]

**Comment by dizzzz** [comment]
Failing tests in linux vms only:

image

image

File: exist-core/pom.xml [hunk 1]
```diff
@@ -1045,7 +1045,12 @@ The BaseX Team. The original license statement is also included below.]]></pream
                         <configuration>
                             <forkCount>1C</forkCount>
                             <reuseForks>false</reuseForks>
-
+                            <argLine>-Dfile.encoding=${project.build.sourceEncoding}</argLine>
```
Comment (adamretter) [inline exist-core/pom.xml:1048]: I don't understand why these changes are needed here or above? I am also concerned that the `log4j.configurationFile` appears to have been lots. Could you explain please?

**Review by adamretter — COMMENTED** [review]
I don't yet understand the purpose of some of the changes. Can we discuss?

File: exist-core/pom.xml (hunk 1 above)
Comment (reinhapa) [reply exist-core/pom.xml:1049]: The mentioned line was dropped by accident and I re-added it in the meantime. The other changes should fixate the test environment in order for calls on methods that return the systems default values will return a predictable result independent on the running system as the environment.

**Comment by dizzzz** [comment]
any status update? @reinhapa @adamretter

**Comment by joewiz** [comment]
@adamretter During the Community Call, we reviewed this PR with an eye toward guessing what questions you might've had with it, and as a result, @reinhapa adjusted the PR to remove the bunch of mvn flags that had been included, and force-pushed it. Now the mvn flags only include the changes necessary to fix the underlying issue; see https://github.com/eXist-db/exist/pull/3669/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L36-R36.

**Review by dizzzz — APPROVED** [review]

**Comment by joewiz** [comment]
@reinhapa Those 4 green checkmarks are a thing of beauty! Thank you for insights to getting CI working again!

**Comment by reinhapa** [comment]
@joewiz you're welcome
